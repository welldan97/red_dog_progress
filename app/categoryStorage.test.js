import {
  index,
  create,
} from './categoryStorage';

import categoryFactory from './categoryFactory';

const mockResponse = (response, { status } = {}) =>
  new Response(JSON.stringify(response), {
    status: status || 200,
    headers: {
      'Content-type': 'application/json',
    },
  });

describe('index', () => {
  beforeEach(() => {
    global.fetch = jest.fn().mockImplementation(() =>
      Promise.resolve(mockResponse([
        categoryFactory(),
        categoryFactory(),
      ])));
  });

  it('It fetches categories', async done => {
    const categories = await index();
    expect(categories).toMatchSnapshot();
    done();
  });
});

describe('create', () => {
  beforeEach(() => {
    global.fetch = jest.fn().mockImplementation(() =>
      Promise.resolve(mockResponse(categoryFactory())));
  });

  it('It creates category', async done => {
    const categories = await create();
    expect(categories).toMatchSnapshot();
    done();
  });
});
