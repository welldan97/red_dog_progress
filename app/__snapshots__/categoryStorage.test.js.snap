// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create It creates category 1`] = `
Object {
  "icon": "@",
  "id": 6203,
  "name": "alarm",
  "type": "counter",
}
`;

exports[`index It fetches categories 1`] = `
Array [
  Object {
    "icon": "@",
    "id": 15266,
    "name": "alarm",
    "type": "counter",
  },
  Object {
    "icon": "@",
    "id": 30235,
    "name": "microchip",
    "type": "counter",
  },
]
`;
